synapse/rest/__init__.py

```diff
@@ -82,55 +82,73 @@ def __init__(self, hs: "HomeServer"):
 
     @staticmethod
     def register_servlets(client_resource: HttpServer, hs: "HomeServer") -> None:
+        # Some servlets are only registered on the main process (and not worker
+        # processes).
+        is_main_process = hs.config.worker.worker_app is None
+
         versions.register_servlets(hs, client_resource)
 
         # Deprecated in r0
-        initial_sync.register_servlets(hs, client_resource)
-        room.register_deprecated_servlets(hs, client_resource)
+        if is_main_process:
```

Comment:
I debated on if we should not do any logic here besides call each `register_servlets` function and have each of those return earlier if the servlet cannot be registered?

WDYT?